test/azure-cli/scenarios.json

```diff
@@ -38,5 +38,15 @@
         "installUsingPython": true
       }
     }
+  },
+  "install_with_python_3_12_bookworm": {
+    "image": "mcr.microsoft.com/devcontainers/python:1-3.12-bookworm",
+    "user": "vscode",
+    "features": {
+      "azure-cli": {
+        "version": "latest",
+        "installUsingPython": true
```

Comment:
Currently, this scenario should fail as `az` does not work when installed with python (that's the issue we are trying to solve here). However, looking at the code in [here](https://github.com/devcontainers/features/blob/26883bd6a77a4f5b14a5012f884803eb0c81a766/src/azure-cli/install.sh#L191-L195), it still installs with `apt` and doesn't attempt to install via `python`. This is definitely a conflict between the experience vs the Feature option. 

I understand that this bug is not due to your PR change, however, can you fix the code? If user wants to explicitly install with python then it should not be installed with `apt`.

Reply:
Once you fix the code, this scenario would start failing. Hence, would be good to update the scenario with `"installUsingPython": false`

Reply:
Defering this to another PR